main.go

```diff
@@ -15,13 +15,504 @@
 package main
 
 import (
+	"context"
+	crand "crypto/rand"
+	"encoding/binary"
+	"fmt"
+	"math/rand"
 	"os"
+	"strconv"
+	"strings"
+
+	"github.com/dolthub/dolt/go/cmd/dolt/cli"
+	"github.com/dolthub/dolt/go/cmd/dolt/commands"
+	"github.com/dolthub/dolt/go/cmd/dolt/commands/sqlserver"
+	eventsapi "github.com/dolthub/dolt/go/gen/proto/dolt/services/eventsapi/v1alpha1"
+	"github.com/dolthub/dolt/go/libraries/doltcore/doltdb"
+	"github.com/dolthub/dolt/go/libraries/doltcore/env"
+	"github.com/dolthub/dolt/go/libraries/doltcore/sqle/dsess"
+	"github.com/dolthub/dolt/go/libraries/events"
+	"github.com/dolthub/dolt/go/libraries/utils/argparser"
+	"github.com/dolthub/dolt/go/libraries/utils/config"
+	"github.com/dolthub/dolt/go/libraries/utils/file"
+	"github.com/dolthub/dolt/go/libraries/utils/filesys"
+	"github.com/dolthub/dolt/go/store/nbs"
+	"github.com/dolthub/dolt/go/store/util/tempfiles"
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/fatih/color"
+	"github.com/tidwall/gjson"
 
 	"github.com/dolthub/doltgresql/server"
 )
 
+var doltgresCommands = cli.NewSubCommandHandler("doltgresql", "it's git for data", []cli.Command{
+	commands.ConfigCmd{},
+	commands.VersionCmd{VersionStr: server.Version},
+	sqlserver.SqlServerCmd{VersionStr: server.Version},
+})
+var globalArgParser = cli.CreateGlobalArgParser("doltgresql")
+
+func init() {
+	events.Application = eventsapi.AppID_APP_DOLTGRES
+
+	if os.Getenv("DOLT_VERBOSE_ASSERT_TABLE_FILES_CLOSED") == "" {
+		nbs.TableIndexGCFinalizerWithStackTrace = false
+	}
+}
+
+const chdirFlag = "--chdir"
+const stdInFlag = "--stdin"
+const stdOutFlag = "--stdout"
+const stdErrFlag = "--stderr"
+const stdOutAndErrFlag = "--out-and-err"
+
 func main() {
-	code, wg := server.RunOnDisk(os.Args[1:])
-	wg.Wait()
-	os.Exit(*code)
+	ctx := context.Background()
+	seedGlobalRand()
+
+	args := os.Args[1:]
+
+	args, err := redirectStdio(args)
+	if err != nil {
+		cli.PrintErrln(err.Error())
+		os.Exit(1)
+	}
+
+	restoreIO := cli.InitIO()
+	defer restoreIO()
+
+	warnIfMaxFilesTooLow()
+
+	fs := filesys.LocalFS
+	dEnv := env.Load(ctx, env.GetCurrentUserHomeDir, fs, doltdb.LocalDirDoltDB, server.Version)
+
+	globalConfig, _ := dEnv.Config.GetConfig(env.GlobalConfig)
+
+	// Inject the "sql-server" command if no other commands were given
+	if len(args) == 0 || (len(args) > 0 && strings.HasPrefix(args[0], "-")) {
+		args = append([]string{"sql-server"}, args...)
+	}
+
+	apr, args, subCommandName, err := parseGlobalArgsAndSubCommandName(globalConfig, args)
+	if err != nil {
+		cli.PrintErrln(err.Error())
+		os.Exit(1)
+	}
+
+	// The sql-server command has special cased logic since it doesn't invoke a Dolt command directly, but runs a server
+	// and waits for it to finish
+	if subCommandName == "" || subCommandName == "sql-server" {
```

Comment:
This should never hit the empty case right? Since we prefix `sql-server` to the args if it's empty?

Reply:
Good catch